Installing-and-Configuring/Flashing-Guides/SPI-Programmer-Best-Practices.md

```diff
@@ -0,0 +1,276 @@
+
+**Where to buy**: 
+- [3mdeb Kit](https://shop.3mdeb.com/shop/modules/ch341a-flash-bios-usb-programmer-kit-soic8-sop8/)
+- [Novacustom Kit](https://novacustom.com/product/modded-ch341a-bios-firmware-programmer-3v/)
+
+### 3. **Raspberry Pi Pico (DIY option)**
+- **Cost**: $4-10 USD
+- **Speed**: ~30 seconds for 16MB chip
+- **Voltage**: 3.3V (configurable via firmware)
+- **Protocols**: SPI (via serprog firmware)
+- **Reliability**: High
+- **Best for**: DIY enthusiasts, custom setups
+
+### 4. **BeagleBone Black (Advanced debugging)**
+- **Cost**: $60-90 USD
+- **Speed**: ~16 seconds for 16MB chip
+- **Voltage**: 3.3V, some pins 1.8V
+- **Additional features**: EHCI debugging, serial console
+- **Best for**: Advanced debugging, development
+
+### 5. **CH347 (Newer alternative to CH341A)**
+- **Cost**: $5-15 USD
+- **Speed**: Significantly faster than CH341A
+- **Voltage**: 3.3V only (5V tolerant inputs)
+- **Protocols**: SPI, I2C
+- **Safety**: Safer than old CH341A (no dangerous 5V output)
+- **Best for**: Budget option with better performance than CH341A
+
+**Note**: The CH347 is a newer alternative to CH341A with proper 3.3V operation and faster speeds. However, it lacks voltage selection for 1.8V chips.
+
+## Programmer Comparison Table
+
+| Programmer | Cost | Speed (16MB) | Voltage Selection | Protocols | Debugging Features | Reliability |
+|------------|------|--------------|-------------------|-----------|-------------------|-------------|
+| **Tigard** | $67-89 | ~42 seconds | 1.8V, 3.3V, 5V, external | SPI, I2C, JTAG, SWD, UART | Logic analyzer, USB debugging | Excellent |
+| **CH341A rev 1.6+** | $5-15 | ~120 seconds | 1.8V, 2.5V, 3.3V, 5V | SPI, I2C | None | Good |
+| **Raspberry Pi Pico** | $4-10 | ~30 seconds | 3.3V (configurable) | SPI | None | High |
+| **BeagleBone Black** | $60-90 | ~16 seconds | 3.3V, 1.8V | SPI, UART | EHCI debugging | High |
+| **CH347** | $5-15 | ~60 seconds | 3.3V only | SPI, I2C | None | Good |
+| **CH341A old** | $2-10 | ~480 seconds | 5V (dangerous) | SPI, I2C | None | **Not recommended** |
+
+## Essential Safety Procedures
+
+### Before Starting
+1. **Power disconnection checklist**:
+   - [ ] Remove laptop battery
+   - [ ] Disconnect AC adapter
+   - [ ] Disconnect CMOS battery (recommended)
+   - [ ] Wait 30 seconds for capacitors to discharge
+
+2. **Workspace preparation**:
+   - Use anti-static wrist strap
+   - Work on anti-static mat
+   - Ensure good lighting
+   - Have magnifying glass available
+
+### Voltage Selection Guidelines
+1. **Start with 1.8V**: Most modern SPI chips prefer 1.8V
+2. **Try 3.3V**: If chip detection fails at 1.8V, try 3.3V
+3. **Never use 5V**: Unless specifically required by chip datasheet
+4. **Test detection**: Run `flashrom -p [programmer]` to verify chip detection
+
+## SPI Chip Pin Mapping (SOIC8)
+
+```
+Chip Pin Layout (top view, dot indicates pin 1):
+┌─────────────┐
```

Comment:
If you want something more graphical, I made a simple SVG that is currently being used in the Skulls documentation.

https://gist.github.com/Thrilleratplay/60dff7cf849de5bdb84d91cabbd4fcfd